js/infobar.js

```diff
@@ -0,0 +1,466 @@
+    // we fallback to the page language.
+    if ($.inArray('translate', bars) > -1) {
+
+        userLangs = userLangs || navigator.languages || navigator.language || navigator.browserLanguage;
+
+        // Note that navigator.language doesn't always work because it's just
+        // the application's locale on some browsers. navigator.languages has
+        // not been widely implemented yet, but the new property provides an
+        // array of the user's accept languages that we'd like to see.
+        // navigator.languages are only supported from Firefox/Chrome 32 and
+        // not at all by IE, Safari, Opera. navigator.language is not supported
+        // by IE so, we need to fallback to navigator.browserLanguage
+        if (navigator.languages) {
+            // Normalize all language strings for easier comparison.
+            acceptLangs = $.map(userLangs, function (lang) {
+                return InfoBar.prototype.normalize(lang);
+            });
+        } else if (navigator.language || navigator.browserLanguage) {
+            // the above properties only return one item but we need
+            // this as an array in the getOfferedLang function.
+            acceptLangs = [userLangs];
+        }
+
+        lang =  InfoBar.prototype.getOfferedLang(acceptLangs) || lang;
+    }
+
+    try {
+        json = JSON.parse(sessionStorage.getItem('infobar.json'));
+    } catch (ex) {
+        console.log('Error thrown while getting json from sessionStorage: ', ex);
+    }
+
+    // check if the JSON exists in sessionStorage
+    if (!json) {
+        var jsonURL = 'https://www.mozilla.org/' + lang + '/infobar/infobar.jsonp';
+         // This loads the data for the transbar which also contains the latest fx version string.
+         $.ajax({
+             url: jsonURL,
+             cache: true,
+             crossDomain: true,
+             dataType: 'jsonp',
+             jsonpCallback: '_',
+             success: function(data) {
+                 // store the returned JSON in sessionStorage to avoid additional Ajax calls
+                 // during this user session.
+                 try {
+                     sessionStorage.setItem('infobar.json', JSON.stringify(data));
+                 } catch(ex) {
+                     console.log('Error while storing JSON in sessionStorage: ', ex);
+                 }
+
+                 json = data;
+                 $(bars).each(function(index, value) {
+                     InfoBar[value].call();
+                 })
+             }
+         }).fail(function(jqXHR, textStatus, errorThrown) {
+             console.log('Error thrown while loading JSON:');
+             console.log('textStatus: ', textStatus);
+             console.log('errorThrown: ', errorThrown);
+         });
+    } else {
+        // the JSON has already been loaded earlier in this session.
+        $(bars).each(function(index, value) {
+            InfoBar[value].call();
+        });
```

Comment:
Small thing, but could factor this `each` loop into a function for a tad more DRY-ness.